src/Control/Session.php

```diff
@@ -135,6 +135,13 @@ class Session
      */
     private static $cookie_name_secure = 'SECSESSID';
 
+    /**
+     * Must be "Strict", "Lax", "None", or empty.
+     * If empty, the default value will be applied.
+     * @config
+     */
+    private static string $cookies_samesite = '';
```

Comment:
I'm kind of inclined to just default set this to 'Lax' here and get rid of the fallback to Cookie::config()->get('default_samesite').  Seems less complex and confusing

Reply:
I'd find it rather confusing as a developer if I changed the default for all cookies to be `Strict` but the session cookie was coming through as `Lax` still. It's a global default for _all_ cookies.

Reply:
> I'd find it rather confusing as a developer if I changed the default for all cookies to be Strict

The default in Cookie.default_samesite = 'Lax' though?

Or do you mean if we were to change this to 'Lax' and a developer left this as 'Lax' but changed Cookie.default_samesite = "Strict" then that would be confusing?  But, even if we left things as is, then a developer can do that anyway since we provide 2 config values

All the confusion surronding this is not a good sign

Is there actually a use case for all thing configurability?  Right now we seem to have:

- Cookie.default_samesite = "Lax"
- Session.cookie_samesite = ""
- CookieJarExtension::updateSameSite extension hook to update the SameSite attribute for individual cookie names

It really feels like there's too much here

Reply:
Yeah I can see that. I'll remove the fallback on `Cookie.default_samesite` for session cookies and update the documentation accordingly.

Reply:
Also possibly better to just get rid of Session.cookie_samesite, rename Cookie.default_same to just Cookie.cookie_samesite.  Do we need 2 different config values here?

Reply:
> Get rid of Session.cookie_samesite

I don't know enough about how cookies are used to really make a call here - but my instinct is that there might be a usecase for "Lax" cookies generally - whereas for security purposes realistically the session cookie should be "Strict". We can't set it as strict by default because that could be a breaking change, but I think giving the ability to configure this separately is useful.

> rename Cookie.default_same to Cookie.cookie_samesite

I'd prefer to keep this name as-is. Even in a major release when we add a `samesite` parameter, I think it would be beneficial to retain this global configuration option (to configure the `samesite` attribute used for cookies in core or in modules), and the "default" especially at that time is important (since it will then be overridable by passing a different value to the parameter).